src/scalems/radical/common.py

```diff
@@ -0,0 +1,236 @@
+
+
+@pytask
+def scalems_dispatch(*args, **kwargs):
+    """SCALE-MS executor for Worker *call* mode.
+
+    Use this (pickled) function as the *function* value in a
+    `radical.pilot.TaskDescription` for raptor-mediated dispatching
+    with `scalems`.
+
+    See Also:
+        `scalems.radical.raptor.dispatch()`
+    """
+    from scalems.radical.raptor import dispatch
+    return dispatch(*args, **kwargs)
+
+
+class _RaptorTaskDescription(typing.TypedDict):
+    """Describe a Task to be executed through a Raptor Worker.
+
+    A specialization of `radical.pilot.TaskDescription`.
+
+    Note the distinctions of a TaskDescription to processed by a raptor.Master.
+
+    The meaning or utility of some fields is dependent on the values of other fields.
+    """
+    uid: str
+    """Unique identifier for the Task across the Session."""
+
+    executable: str
+    """Unused by Raptor tasks."""
+
+    scheduler: str
+    """The UID of the raptor.Master scheduler task."""
+
+    mode: str
+    """The executor mode for the Worker to use.
+
+    For ``call`` mode, either *function* or *method* must name a task executor.
+    Depending on the Worker (sub)class, resources such as an `mpi4py.MPI.Comm`
+    will be provided as the first positional argument to the executor.
+    ``*args`` and ``**kwargs`` will be provided to the executor from the corresponding
+    fields.
+    """
+
+    function: str
+    """For 'call' mode, a callable for dispatching.
+
+    The callable can either be a function object present in the namespace of the interpreter launched
+    by the Worker for the task, or a `radical.pilot.pytask` pickled function object.
+    """
+
+    method: str
+    """For 'call' mode, a member function of the Worker for dispatching."""
+
+    args: list
+    """For 'call' mode, list of positional arguments to provide to the executor method or function."""
+
+    kwargs: dict
+    """For 'call' mode, a dictionary of key word arguments to provide to the executor method or function."""
+
+
+class TaskDictionary(typing.TypedDict):
+    """Task representations seen by *request_cb* and *result_cb*.
+
+    Other fields may be present, but the objects in the sequences provided to
+    :py:meth:`scalems.radical.raptor.ScaleMSMaster.request_cb()` and
+    :py:meth:`scalems.radical.raptor.ScaleMSMaster.result_cb()` have the following fields.
+    Result fields will not be populated until the Task runs.
+    """
+    uid: str
+    """Canonical identifier for the Task.
+
+    Note that *uid* may be omitted from the original TaskDescription.
+    """
+
+    description: _RaptorTaskDescription
+    """Encoding of the original task description."""
+
+    out: str
+    """Task standard output."""
+
+    err: str
+    """Task standard error."""
+
+    ret: int
+    """Function return code."""
+
+    val: typing.Any
+    """Function return value."""
+
+    exc: typing.Tuple[str, str]
+    """Exception type name and message."""
```

Comment:
All of these field names are wrong, according to https://radicalpilot.readthedocs.io/en/stable/_modules/radical/pilot/task.html#Task.as_dict